### PR TITLE
API-Based Web App

### PR DESCRIPTION
**Pull request by roodjinocherilus**
In his PR, we have created:
- a home page that lists movies
- show and add likes
- A comment page that shows detail of the movies and lists comments
- A form to add new comments
- A reservation page to show the reservation
- Add new reservation
- retrieve and add data from TVMaze API
- retrieve and add data from involvement API

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,5 @@
-import _ from 'lodash';
-import printMe from './print.js';
+// import _ from 'lodash';
 import './style.css';
```
Comment (TadesseDev) [inline src/index.js:2]: - [ ] Please consider removing commented codes as it will make your codebase messy, Remember if you ever like to see the code you have written once you can use your commit history.

File: src/modules/homePage/load-homepage.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <img src="${response.image.medium}" alt="movie posture">
+  </div>
+  <div class="row-2">
+    <h3 class="title-movie">${response.name}</h3>
+    <p class="flex-col">
+      <button class="heart">
+        <i class="fa fa-heart heart-icon" id="${response.id}"></i>
+      </button><span class=like-count>0</span>likes
+    </p>
+  </div>
```
Comment (TadesseDev) [inline src/modules/homePage/load-homepage.js:25]: - [ ] Good job team. Please notice here the like counter should not be static, I notice you are correct in sendiong an update to your involvement API when I click the like button. But when the page reloads the changes are lost. Please make sure when the user clicks on the like button the data is recorded inside the involvement API and when we reload the page we gate the same amount of likes we already make.

File: src/modules/comment/popup.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  }
+};
+
+const closeModal = () => {
+  const modalContainer = document.getElementById('modal-container');
+  const modal = document.querySelector('.modal');
+  const backdrop = document.querySelector('.modal-backdrop');
+  modalContainer.removeChild(modal);
+  modalContainer.removeChild(backdrop);
+};
```
Comment (TadesseDev) [inline src/modules/comment/popup.js:15]: #### Great job on your comment popup team. But I suggest you do the following. 
- [ ] When a user clicks on the "Comment" button, beside the data being recorded to the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270), the screen should be updated automatically with the new comment.
- [ ] Also, Please add a visible `close` button or icon. Currently, I had to double-click on the free space to get rid of the pop-up. this is against the wireframe and not good when it comes to UX.
- [ ] There should be a comment counter on the pop-up window, and the value should be incremented accordingly. when we add a new comment the comment count should be incremented and the screen is populated with the new count. and remember, the counter should be implemented as a separate module.
- [ ] Once we have the comment counter, we need to cover the counter function with unit tests using Jest. Make sure that you have tests for all edge cases

https://user-images.githubusercontent.com/69077061/221371093-de276cbc-974e-4119-90de-513642e7f44c.mp4

File: src/modules/reserve/getReserverations.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const getReserverations = async (movieId) => {
+  const request = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/sUHYcv06blVrKgUc2kUH/reservations?item_id=${movieId}`);
+  const response = await request.json();
+
+  return response;
+};
```
Comment (TadesseDev) [inline src/modules/reserve/getReserverations.js:6]: #### Great job on your reservation popup team. But I suggest you address the following. 
- [ ] When a user clicks on the "Reserve" button, beside the data being recorded in the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270), the screen should be updated as a result with the new reservation we make.
- [ ] Your Reservation seems to be falling for some of the movies. Please make sure it's possible to reserve any of the movies that we have listed.
- [ ] Also, As the wire frame suggested the pop-up better be centered on the page.
- [ ] I suggest adding a reservation counter on the pop-up window, and the value should be incremented accordingly. when we add a new reservation the reservation count should be incremented and the screen is populated with the new count. Remember, The counter should be implemented as a separate module.
- [ ] Once we have the reservation counter, we need to cover the counter function with unit tests using Jest. Make sure that you have tests for all edge cases.

https://user-images.githubusercontent.com/69077061/221371435-3011de83-9da5-43da-a267-10d7f7f50214.mp4

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team
### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7

## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (FevenSeyfu) [reply src/index.js:2]: thank you for the feedback removed the comment

File: src/modules/homePage/load-homepage.js (hunk 2 above)
Comment (FevenSeyfu) [reply src/modules/homePage/load-homepage.js:25]: Retain number of likes when page reloads.

File: src/index.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply src/index.js]: Fixed.

File: src/modules/comment/popup.js (hunk 3 above)
Comment (Abenezer-Tilahun) [reply src/modules/comment/popup.js]: Fixed.

File: src/modules/reserve/getReserverations.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const getReserverations = async (movieId) => {
+  const request = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/sUHYcv06blVrKgUc2kUH/reservations?item_id=${movieId}`);
+  const response = await request.json();
+  return response;
+};
+
+export default getReserverations;
```
Comment (Abenezer-Tilahun) [inline src/modules/reserve/getReserverations.js:7]: - [ ] Good Job for Implementing some of the point the previous reviewer suggest just like possible to reserve any of the movies that we have listed, As the wire frame suggested the pop-up  centered on the page .
- [ ] But Kindly I suggest you to implemented the following point which should be addressed.
- When a user clicks on the "Reserve" button the screen  updated a result with the new reservation we make but it replace the First one with the new reservation I make Just like the video below.
- I suggest adding a reservation counter on the pop-up window, and the value should be incremented accordingly. when we add a new reservation the reservation count should be incremented and the screen is populated with the new count. Remember, The counter should be implemented as a separate module.

Please check this [video](https://drive.google.com/file/d/1a5HZ77WRuXSJoCYtksimVOIpzSCeuDEN/view?usp=share_link)

File: src/modules/comment/CommentCount.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    const count = CommentCount();
+    expect(count).toBe(0);
+  });
+
+  test('returns NaN if the total comments element is not found', () => {
+    document.body.innerHTML = '<div></div>';
+    const count = CommentCount();
+    expect(Number.isNaN(count)).toBe(true);
+  });
+});
```
Comment (Abenezer-Tilahun) [inline src/modules/comment/CommentCount.test.js:33]: - [ ] As stated in the project requirement These count functions need to be covered with unit tests using Jest and should be passed , So kindly make sure that All test are passing.
Capture

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @roodjinocherilus  @FevenSeyfu @Eldonratzlaff  , 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 👏🏻

✔️ No Linters error
✔️ You used correct Git Hub-Flow.
✔️ Good commit messages
✔️ Professional READEME 


almost

## Required Changes ♻️
_Check the comments below the review._

- [ ] As stated in the project requirement  For Reservation  the API gives you reserve count numbers, **So kindly  create a specific function to calculate these numbers in each page. These count functions need to be covered with unit tests using Jest.**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/comment/CommentCount.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const CommentCount = () => {
+  const totalComments = document.getElementById('items');
```
Comment (1GORDON) [inline src/modules/comment/CommentCount.js:2]: Great job team 🤝

- [ ] As the previous reviewers suggested, kindly make sure the comments counter updates instantly once a new comment is added. Right now the counter is not changing.

File: src/modules/reserve/countreserve.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import getReserverations from './getReserverations.js';
+
+const countReserve = async (id) => {
```
Comment (1GORDON) [inline src/modules/reserve/countreserve.js:3]: Great job team🤝

- [ ] As the previous reviewers suggested, kindly add a reservations counter on your popup and make sure this reservations counter is updated every time a new reservation is submitted.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi team ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/reserve/countreserve.js (hunk 8 above)
Comment (Davidosky007) [reply src/modules/reserve/countreserve.js:3]: * Kindly add tests to cover all your count functions as its a requirement for your capstone project 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @roodjinocherilus @FevenSeyfu @Abenezer-Tilahun 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
